File: sistema_financeiro/src/routes/contas.py
from flask import Blueprint, request, jsonify
from datetime import datetime, timedelta
from src.main import db
from src.models.conta import Conta, Periodicidade
from src.models.categoria import Categoria
from src.models.transacao import Transacao

contas_bp = Blueprint('contas', __name__)

@contas_bp.route('/', methods=['GET'])
def listar_contas():
    """Lista contas com filtros opcionais."""
    # Parâmetros de filtro
    status = request.args.get('status')
    fixa = request.args.get('fixa')
    categoria_id = request.args.get('categoria_id')
    
    # Construir a query base
    query = Conta.query
    
    # Aplicar filtros
    if status:
        query = query.filter(Conta.status == status)
    
    if fixa is not None:
        fixa_bool = fixa.lower() == 'true'
        query = query.filter(Conta.fixa == fixa_bool)
    
    if categoria_id:
        query = query.filter(Conta.categoria_id == categoria_id)
    
    # Ordenar por data de vencimento
    contas = query.order_by(Conta.data_vencimento).all()
    
    return jsonify({
        'status': 'success',
        'data': [conta.to_dict() for conta in contas]
    })

@contas_bp.route('/', methods=['POST'])
def criar_conta():
    """Cria uma nova conta."""
    dados = request.json
    
    # Validar campos obrigatórios
    campos_obrigatorios = ['nome', 'valor', 'categoria_id']
    for campo in campos_obrigatorios:
        if campo not in dados:
            return jsonify({
                'status': 'error',
                'message': f'Campo {campo} é obrigatório'
            }), 400
    
    # Validar categoria
    categoria = Categoria.query.get(dados['categoria_id'])
    if not categoria:
        return jsonify({
            'status': 'error',
            'message': 'Categoria não encontrada'
        }), 404
    
    # Processar a data de vencimento
    data_vencimento = None
    if 'data_vencimento' in dados and dados['data_vencimento']:
        try:
            data_vencimento = datetime.strptime(dados['data_vencimento'], '%Y-%m-%d').date()
        except ValueError:
            return jsonify({
                'status': 'error',
                'message': 'Formato de data inválido. Use YYYY-MM-DD'
            }), 400
    
    # Processar periodicidade
    periodicidade = None
    if 'periodicidade' in dados and dados['periodicidade']:
        try:
            periodicidade = Periodicidade(dados['periodicidade'])
        except ValueError:
            return jsonify({
                'status': 'error',
                'message': f'Periodicidade inválida. Valores válidos: {", ".join([p.value for p in Periodicidade])}'
            }), 400
    
    # Criar a conta
    nova_conta = Conta(
        nome=dados['nome'],
        valor=float(dados['valor']),
        data_vencimento=data_vencimento,
        fixa=dados.get('fixa', False),
        periodicidade=periodicidade,
        status=dados.get('status', 'pendente'),
        observacoes=dados.get('observacoes', ''),
        categoria_id=dados['categoria_id']
    )
    
    db.session.add(nova_conta)
    db.session.commit()
    
    return jsonify({
        'status': 'success',
        'message': 'Conta registrada com sucesso',
        'data': nova_conta.to_dict()
    }), 201

@contas_bp.route('/<int:id>', methods=['PUT'])
def atualizar_conta(id):
    """Atualiza uma conta existente."""
    conta = Conta.query.get(id)
    
    if not conta:
        return jsonify({
            'status': 'error',
            'message': 'Conta não encontrada'
        }), 404
    
    dados = request.json
    
    if not dados:
        return jsonify({
            'status': 'error',
            'message': 'Dados inválidos'
        }), 400
    
    # Atualizar campos
    if 'nome' in dados:
        conta.nome = dados['nome']
    
    if 'valor' in dados:
        conta.valor = float(dados['valor'])
    
    if 'data_vencimento' in dados:
        if dados['data_vencimento']:
            try:
                conta.data_vencimento = datetime.strptime(dados['data_vencimento'], '%Y-%m-%d').date()
            except ValueError:
                return jsonify({
                    'status': 'error',
                    'message': 'Formato de data inválido. Use YYYY-MM-DD'
                }), 400
        else:
            conta.data_vencimento = None
    
    if 'fixa' in dados:
        conta.fixa = dados['fixa']
    
    if 'periodicidade' in dados:
        if dados['periodicidade']:
            try:
                conta.periodicidade = Periodicidade(dados['periodicidade'])
            except ValueError:
                return jsonify({
                    'status': 'error',
                    'message': f'Periodicidade inválida. Valores válidos: {", ".join([p.value for p in Periodicidade])}'
                }), 400
        else:
            conta.periodicidade = None
    
    if 'status' in dados:
        conta.status = dados['status']
    
    if 'observacoes' in dados:
        conta.observacoes = dados['observacoes']
    
    if 'categoria_id' in dados:
        categoria = Categoria.query.get(dados['categoria_id'])
        if not categoria:
            return jsonify({
                'status': 'error',
                'message': 'Categoria não encontrada'
            }), 404
        conta.categoria_id = dados['categoria_id']
    
    db.session.commit()
    
    return jsonify({
        'status': 'success',
        'message': 'Conta atualizada com sucesso',
        'data': conta.to_dict()
    })

@contas_bp.route('/<int:id>', methods=['DELETE'])
def excluir_conta(id):
    """Exclui uma conta existente."""
    conta = Conta.query.get(id)
    
    if not conta:
        return jsonify({
            'status': 'error',
            'message': 'Conta não encontrada'
        }), 404
    
    db.session.delete(conta)
    db.session.commit()
    
    return jsonify({
        'status': 'success',
        'message': 'Conta excluída com sucesso'
    })

@contas_bp.route('/<int:id>/pagar', methods=['POST'])
def pagar_conta(id):
    """Marca uma conta como paga e cria uma transação correspondente."""
    conta = Conta.query.get(id)
    
    if not conta:
        return jsonify({
            'status': 'error',
            'message': 'Conta não encontrada'
        }), 404
    
    if conta.status == 'paga':
        return jsonify({
            'status': 'error',
            'message': 'Esta conta já está paga'
        }), 400
    
    dados = request.json or {}
    
    # Data do pagamento (padrão: hoje)
    data_pagamento = datetime.now().date()
    if 'data_pagamento' in dados:
        try:
            data_pagamento = datetime.strptime(dados['data_pagamento'], '%Y-%m-%d').date()
        except ValueError:
            return jsonify({
                'status': 'error',
                'message': 'Formato de data inválido. Use YYYY-MM-DD'
            }), 400
    
    # Criar transação de saída
    transacao = Transacao(
        data=data_pagamento,
        valor=conta.valor,
        descricao=f"Pagamento: {conta.nome}",
        tipo='saída',
        categoria_id=conta.categoria_id,
        fixa=conta.fixa
    )
    
    db.session.add(transacao)
    
    # Atualizar conta
    conta.status = 'paga'
    conta.transacao_id = transacao.id
    
    # Se for conta fixa, criar próxima conta
    if conta.fixa and conta.periodicidade:
        proxima_data = None
        
        if conta.periodicidade == Periodicidade.DIARIA:
            proxima_data = conta.data_vencimento + timedelta(days=1)
        elif conta.periodicidade == Periodicidade.SEMANAL:
            proxima_data = conta.data_vencimento + timedelta(weeks=1)
        elif conta.periodicidade == Periodicidade.QUINZENAL:
            proxima_data = conta.data_vencimento + timedelta(days=15)
        elif conta.periodicidade == Periodicidade.MENSAL:
            # Adicionar um mês (considerando o último dia do mês)
            mes = conta.data_vencimento.month + 1
            ano = conta.data_vencimento.year
            if mes > 12:
                mes = 1
                ano += 1
            
            dia = min(conta.data_vencimento.day, [31, 29 if ano % 4 == 0 and (ano % 100 != 0 or ano % 400 == 0) else 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][mes-1])
            proxima_data = datetime(ano, mes, dia).date()
        elif conta.periodicidade == Periodicidade.BIMESTRAL:
            mes = conta.data_vencimento.month + 2
            ano = conta.data_vencimento.year
            if mes > 12:
                mes -= 12
                ano += 1
            
            dia = min(conta.data_vencimento.day, [31, 29 if ano % 4 == 0 and (ano % 100 != 0 or ano % 400 == 0) else 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][mes-1])
            proxima_data = datetime(ano, mes, dia).date()
        elif conta.periodicidade == Periodicidade.TRIMESTRAL:
            mes = conta.data_vencimento.month + 3
            ano = conta.data_vencimento.year
            if mes > 12:
                mes -= 12
                ano += 1
            
            dia = min(conta.data_vencimento.day, [31, 29 if ano % 4 == 0 and (ano % 100 != 0 or ano % 400 == 0) else 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][mes-1])
            proxima_data = datetime(ano, mes, dia).date()
        elif conta.periodicidade == Periodicidade.SEMESTRAL:
            mes = conta.data_vencimento.month + 6
            ano = conta.data_vencimento.year
            if mes > 12:
                mes -= 12
                ano += 1
            
            dia = min(conta.data_vencimento.day, [31, 29 if ano % 4 == 0 and (ano % 100 != 0 or ano % 400 == 0) else 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31][mes-1])
            proxima_data = datetime(ano, mes, dia).date()
        elif conta.periodicidade == Periodicidade.ANUAL:
            proxima_data = datetime(conta.data_vencimento.year + 1, conta.data_vencimento.month, conta.data_vencimento.day).date()
        
        if proxima_data:
            nova_conta = Conta(
                nome=conta.nome,
                valor=conta.valor,
                data_vencimento=proxima_data,
                fixa=True,
                periodicidade=conta.periodicidade,
                status='pendente',
                observacoes=conta.observacoes,
                categoria_id=conta.categoria_id
            )
            db.session.add(nova_conta)
    
    db.session.commit()
    
    return jsonify({
        'status': 'success',
        'message': 'Conta marcada como paga com sucesso',
        'data': {
            'conta': conta.to_dict(),
            'transacao': transacao.to_dict()
        }
    })

@contas_bp.route('/vencidas', methods=['GET'])
def listar_contas_vencidas():
    """Lista contas vencidas."""
    hoje = datetime.now().date()
    
    contas_vencidas = Conta.query.filter(
        Conta.data_vencimento < hoje,
        Conta.status == 'pendente'
    ).order_by(Conta.data_vencimento).all()
    
    return jsonify({
        'status': 'success',
        'data': [conta.to_dict() for conta in contas_vencidas]
    })

@contas_bp.route('/proximas', methods=['GET'])
def listar_proximas_contas():
    """Lista contas a vencer nos próximos dias."""
    hoje = datetime.now().date()
    dias = request.args.get('dias', 7, type=int)
    
    data_limite = hoje + timedelta(days=dias)
    
    proximas_contas = Conta.query.filter(
        Conta.data_vencimento >= hoje,
        Conta.data_vencimento <= data_limite,
        Conta.status == 'pendente'
    ).order_by(Conta.data_vencimento).all()
    
    return jsonify({
        'status': 'success',
        'data': [conta.to_dict() for conta in proximas_contas]
    })

File: sistema_financeiro/src/static/js/contas.js
// Contas.js - Script para a página de contas

document.addEventListener('DOMContentLoaded', function() {
    // Inicializar data atual no formulário
    document.getElementById('conta-vencimento').valueAsDate = new Date();
    document.getElementById('pagar-conta-data').valueAsDate = new Date();
    
    // Carregar dados iniciais
    carregarCategorias();
    carregarContas();
    carregarContasVencidas();
    carregarProximasContas();
    
    // Configurar eventos
    document.getElementById('form-filtros').addEventListener('submit', function(e) {
        e.preventDefault();
        carregarContas();
    });
    
    document.getElementById('btn-limpar-filtros').addEventListener('click', limparFiltros);
    document.getElementById('btn-salvar-conta').addEventListener('click', salvarConta);
    document.getElementById('btn-confirmar-pagamento').addEventListener('click', pagarConta);
    
    // Mostrar/ocultar campo de periodicidade quando conta fixa for marcada/desmarcada
    document.getElementById('conta-fixa').addEventListener('change', function() {
        document.getElementById('div-periodicidade').style.display = this.checked ? 'block' : 'none';
        
        // Limpar campo de periodicidade quando desmarcado
        if (!this.checked) {
            document.getElementById('conta-periodicidade').value = '';
        }
    });
    
    // Configurar eventos de abas
    document.getElementById('vencidas-tab').addEventListener('click', function() {
        carregarContasVencidas();
    });
    
    document.getElementById('proximas-tab').addEventListener('click', function() {
        carregarProximasContas();
    });
});

// Função para formatar valores monetários
function formatarMoeda(valor) {
    return new Intl.NumberFormat('pt-BR', {
        style: 'currency',
        currency: 'BRL'
    }).format(valor);
}

// Função para formatar datas
function formatarData(dataStr) {
    if (!dataStr) return '-';
    const data = new Date(dataStr);
    return data.toLocaleDateString('pt-BR');
}

// Carregar categorias
function carregarCategorias() {
    fetch('/api/categorias/?tipo=saída')
        .then(response => response.json())
        .then(data => {
            if (data.status === 'success') {
                const categorias = data.data;
                const selectFiltro = document.getElementById('filtro-categoria');
                const selectConta = document.getElementById('conta-categoria');
                
                // Limpar opções existentes
                selectFiltro.innerHTML = '<option value="">Todas</option>';
                selectConta.innerHTML = '<option value="">Selecione uma categoria</option>';
                
                // Adicionar categorias
                categorias.forEach(categoria => {
                    // Para o filtro
                    const optionFiltro = document.createElement('option');
                    optionFiltro.value = categoria.id;
                    optionFiltro.textContent = categoria.nome;
                    selectFiltro.appendChild(optionFiltro);
                    
                    // Para o formulário de conta
                    const optionConta = document.createElement('option');
                    optionConta.value = categoria.id;
                    optionConta.textContent = categoria.nome;
                    selectConta.appendChild(optionConta);
                });
            }
        })
        .catch(error => console.error('Erro ao carregar categorias:', error));
}

// Carregar contas
function carregarContas() {
    // Obter filtros
    const status = document.getElementById('filtro-status').value;
    const tipo = document.getElementById('filtro-tipo').value;
    const categoriaId = document.getElementById('filtro-categoria').value;
    
    // Construir URL com filtros
    let url = '/api/contas/?';
    if (status) url += `status=${status}&`;
    if (tipo) url += `fixa=${tipo}&`;
    if (categoriaId) url += `categoria_id=${categoriaId}&`;
    
    fetch(url)
        .then(response => response.json())
        .then(data => {
            if (data.status === 'success') {
                const contas = data.data;
                const tbody = document.getElementById('lista-contas');
                
                if (contas.length === 0) {
                    tbody.innerHTML = '<tr><td colspan="7" class="text-center">Nenhuma conta encontrada</td></tr>';
                    return;
                }
                
                tbody.innerHTML = '';
                
                contas.forEach(conta => {
                    const tr = document.createElement('tr');
                    
                    // Data de vencimento
                    const tdData = document.createElement('td');
                    tdData.textContent = formatarData(conta.data_vencimento);
                    tr.appendChild(tdData);
                    
                    // Nome
                    const tdNome = document.createElement('td');
                    tdNome.textContent = conta.nome;
                    tr.appendChild(tdNome);
                    
                    // Categoria
                    const tdCat = document.createElement('td');
                    tdCat.textContent = conta.categoria_nome;
                    tr.appendChild(tdCat);
                    
                    // Valor
                    const tdValor = document.createElement('td');
                    tdValor.textContent = formatarMoeda(conta.valor);
                    tdValor.classList.add('valor-saida');
                    tr.appendChild(tdValor);
                    
                    // Status
                    const tdStatus = document.createElement('td');
                    const spanStatus = document.createElement('span');
                    spanStatus.classList.add(`status-${conta.status}`);
                    spanStatus.textContent = conta.status.charAt(0).toUpperCase() + conta.status.slice(1);
                    tdStatus.appendChild(spanStatus);
                    tr.appendChild(tdStatus);
                    
                    // Tipo
                    const tdTipo = document.createElement('td');
                    tdTipo.textContent = conta.fixa ? 'Fixa' : 'Variável';
                    if (conta.fixa && conta.periodicidade) {
                        tdTipo.textContent += ` (${conta.periodicidade})`;
                    }
                    tr.appendChild(tdTipo);
                    
                    // Ações
                    const tdAcoes = document.createElement('td');
                    
                    // Botão Pagar (apenas para contas pendentes)
                    if (conta.status === 'pendente') {
                        const btnPagar = document.createElement('button');
                        btnPagar.classList.add('btn', 'btn-sm', 'btn-success', 'me-1');
                        btnPagar.innerHTML = '<i class="bi bi-check-circle"></i>';
                        btnPagar.title = 'Pagar Conta';
                        btnPagar.onclick = function() {
                            prepararPagamento(conta);
                        };
                        tdAcoes.appendChild(btnPagar);
                    }
                    
                    // Botão Editar
                    const btnEditar = document.createElement('button');
                    btnEditar.classList.add('btn', 'btn-sm', 'btn-primary', 'me-1');
                    btnEditar.innerHTML = '<i class="bi bi-pencil"></i>';
                    btnEditar.title = 'Editar';
                    btnEditar.onclick = function() {
                        editarConta(conta);
                    };
                    tdAcoes.appendChild(btnEditar);
                    
                    // Botão Excluir
                    const btnExcluir = document.createElement('button');
                    btnExcluir.classList.add('btn', 'btn-sm', 'btn-danger');
                    btnExcluir.innerHTML = '<i class="bi bi-trash"></i>';
                    btnExcluir.title = 'Excluir';
                    btnExcluir.onclick = function() {
                        confirmarExclusao(conta.id);
                    };
                    tdAcoes.appendChild(btnExcluir);
                    
                    tr.appendChild(tdAcoes);
                    
                    tbody.appendChild(tr);
                });
            }
        })
        .catch(error => console.error('Erro ao carregar contas:', error));
}

// Carregar contas vencidas
function carregarContasVencidas() {
    fetch('/api/contas/vencidas')
        .then(response => response.json())
        .then(data => {
            if (data.status === 'success') {
                const contas = data.data;
                const tbody = document.getElementById('lista-contas-vencidas');
                
                // Atualizar contador
                document.getElementById('contador-vencidas').textContent = contas.length;
                
                if (contas.length === 0) {
                    tbody.innerHTML = '<tr><td colspan="5" class="text-center">Nenhuma conta vencida</td></tr>';
                    return;
                }
                
                tbody.innerHTML = '';
                
                contas.forEach(conta => {
                    const tr = document.createElement('tr');
                    
                    // Data de vencimento
                    const tdData = document.createElement('td');
                    tdData.textContent = formatarData(conta.data_vencimento);
                    tr.appendChild(tdData);
                    
                    // Nome
                    const tdNome = document.createElement('td');
                    tdNome.textContent = conta.nome;
                    tr.appendChild(tdNome);
                    
                    // Categoria
                    const tdCat = document.createElement('td');
                    tdCat.textContent = conta.categoria_nome;
                    tr.appendChild(tdCat);
                    
                    // Valor
                    const tdValor = document.createElement('td');
                    tdValor.textContent = formatarMoeda(conta.valor);
                    tdValor.classList.add('valor-saida');
                    tr.appendChild(tdValor);
                    
                    // Ações
                    const tdAcoes = document.createElement('td');
                    
                    // Botão Pagar
                    const btnPagar = document.createElement('button');
                    btnPagar.classList.add('btn', 'btn-sm', 'btn-success', 'me-1');
                    btnPagar.innerHTML = '<i class="bi bi-check-circle"></i>';
                    btnPagar.title = 'Pagar Conta';
                    btnPagar.onclick = function() {
                        prepararPagamento(conta);
                    };
                    tdAcoes.appendChild(btnPagar);
                    
                    tr.appendChild(tdAcoes);
                    
                    tbody.appendChild(tr);
                });
            }
        })
        .catch(error => console.error('Erro ao carregar contas vencidas:', error));
}

// Carregar próximas contas
function carregarProximasContas() {
    fetch('/api/contas/proximas?dias=7')
        .then(response => response.json())
        .then(data => {
            if (data.status === 'success') {
                const contas = data.data;
                const tbody = document.getElementById('lista-proximas-contas');
                
                // Atualizar contador
                document.getElementById('contador-proximas').textContent = contas.length;
                
                if (contas.length === 0) {
                    tbody.innerHTML = '<tr><td colspan="5" class="text-center">Nenhuma conta próxima do vencimento</td></tr>';
                    return;
                }
                
                tbody.innerHTML = '';
                
                contas.forEach(conta => {
                    const tr = document.createElement('tr');
                    
                    // Data de vencimento
                    const tdData = document.createElement('td');
                    tdData.textContent = formatarData(conta.data_vencimento);
                    tr.appendChild(tdData);
                    
                    // Nome
                    const tdNome = document.createElement('td');
                    tdNome.textContent = conta.nome;
                    tr.appendChild(tdNome);
                    
                    // Categoria
                    const tdCat = document.createElement('td');
                    tdCat.textContent = conta.categoria_nome;
                    tr.appendChild(tdCat);
                    
                    // Valor
                    const tdValor = document.createElement('td');
                    tdValor.textContent = formatarMoeda(conta.valor);
                    tdValor.classList.add('valor-saida');
                    tr.appendChild(tdValor);
                    
                    // Ações
                    const tdAcoes = document.createElement('td');
                    
                    // Botão Pagar
                    const btnPagar = document.createElement('button');
                    btnPagar.classList.add('btn', 'btn-sm', 'btn-success', 'me-1');
                    btnPagar.innerHTML = '<i class="bi bi-check-circle"></i>';
                    btnPagar.title = 'Pagar Conta';
                    btnPagar.onclick = function() {
                        prepararPagamento(conta);
                    };
                    tdAcoes.appendChild(btnPagar);
                    
                    tr.appendChild(tdAcoes);
                    
                    tbody.appendChild(tr);
                });
            }
        })
        .catch(error => console.error('Erro ao carregar próximas contas:', error));
}

// Limpar filtros
function limparFiltros() {
    document.getElementById('filtro-status').value = '';
    document.getElementById('filtro-tipo').value = '';
    document.getElementById('filtro-categoria').value = '';
    
    carregarContas();
}

// Salvar conta
function salvarConta() {
    const id = document.getElementById('conta-id').value;
    const nome = document.getElementById('conta-nome').value;
    const valor = document.getElementById('conta-valor').value;
    const dataVencimento = document.getElementById('conta-vencimento').value;
    const categoriaId = document.getElementById('conta-categoria').value;
    const fixa = document.getElementById('conta-fixa').checked;
    const periodicidade = fixa ? document.getElementById('conta-periodicidade').value : null;
    const observacoes = document.getElementById('conta-observacoes').value;
    
    // Validar campos obrigatórios
    if (!nome || !valor || !dataVencimento || !categoriaId) {
        alert('Por favor, preencha todos os campos obrigatórios.');
        return;
    }
    
    // Validar periodicidade para contas fixas
    if (fixa && !periodicidade) {
        alert('Por favor, selecione a periodicidade para contas fixas.');
        return;
    }
    
    const conta = {
        nome,
        valor: parseFloat(valor),
        data_vencimento: dataVencimento,
        categoria_id: parseInt(categoriaId),
        fixa,
        periodicidade,
        observacoes,
        status: 'pendente'
    };
    
    // Determinar se é uma criação ou atualização
    const metodo = id ? 'PUT' : 'POST';
    const url = id ? `/api/contas/${id}` : '/api/contas/';
    
    fetch(url, {
        method: metodo,
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(conta)
    })
    .then(response => response.json())
    .then(data => {
        if (data.status === 'success') {
            // Fechar modal e recarregar contas
            const modal = bootstrap.Modal.getInstance(document.getElementById('modalConta'));
            modal.hide();
            
            carregarContas();
            carregarContasVencidas();
            carregarProximasContas();
            limparFormulario();
            
            alert(id ? 'Conta atualizada com sucesso!' : 'Conta registrada com sucesso!');
        } else {
            alert('Erro: ' + data.message);
        }
    })
    .catch(error => {
        console.error('Erro ao salvar conta:', error);
        alert('Ocorreu um erro ao salvar a conta.');
    });
}

// Preparar pagamento de conta
function prepararPagamento(conta) {
    document.getElementById('pagar-conta-id').value = conta.id;
    document.getElementById('pagar-conta-nome').value = conta.nome;
    document.getElementById('pagar-conta-valor').value = formatarMoeda(conta.valor);
    
    // Abrir modal
    const modal = new bootstrap.Modal(document.getElementById('modalPagarConta'));
    modal.show();
}

// Pagar conta
function pagarConta() {
    const contaId = document.getElementById('pagar-conta-id').value;
    const dataPagamento = document.getElementById('pagar-conta-data').value;
    
    // Validar data
    if (!dataPagamento) {
        alert('Por favor, informe a data de pagamento.');
        return;
    }
    
    fetch(`/api/contas/${contaId}/pagar`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({
            data_pagamento: dataPagamento
        })
    })
    .then(response => response.json())
    .then(data => {
        if (data.status === 'success') {
            // Fechar modal e recarregar contas
            const modal = bootstrap.Modal.getInstance(document.getElementById('modalPagarConta'));
            modal.hide();
            
            carregarContas();
            carregarContasVencidas();
            carregarProximasContas();
            
            alert('Conta paga com sucesso!');
        } else {
            alert('Erro: ' + data.message);
        }
    })
    .catch(error => {
        console.error('Erro ao pagar conta:', error);
        alert('Ocorreu um erro ao pagar a conta.');
    });
}

// Editar conta
function editarConta(conta) {
    // Preencher formulário com dados da conta
    document.getElementById('conta-id').value = conta.id;
    document.getElementById('conta-nome').value = conta.nome;
    document.getElementById('conta-valor').value = conta.valor;
    document.getElementById('conta-categoria').value = conta.categoria_id;
    document.getElementById('conta-observacoes').value = conta.observacoes || '';
    
    // Data de vencimento
    if (conta.data_vencimento) {
        document.getElementById('conta-vencimento').value = conta.data_vencimento.split('T')[0];
    }
    
    // Conta fixa e periodicidade
    document.getElementById('conta-fixa').checked = conta.fixa;
    document.getElementById('div-periodicidade').style.display = conta.fixa ? 'block' : 'none';
    
    if (conta.fixa && conta.periodicidade) {
        document.getElementById('conta-periodicidade').value = conta.periodicidade;
    } else {
        document.getElementById('conta-periodicidade').value = '';
    }
    
    // Atualizar título do modal
    document.getElementById('modalContaLabel').textContent = 'Editar Conta';
    
    // Abrir modal
    const modal = new bootstrap.Modal(document.getElementById('modalConta'));
    modal.show();
}

// Confirmar exclusão
function confirmarExclusao(id) {
    // Armazenar ID da conta a ser excluída
    document.getElementById('btn-confirmar-exclusao').dataset.id = id;
    
    // Configurar evento de exclusão
    document.getElementById('btn-confirmar-exclusao').onclick = function() {
        const contaId = this.dataset.id;
        excluirConta(contaId);
    };
    
    // Abrir modal de confirmação
    const modal = new bootstrap.Modal(document.getElementById('modalConfirmacao'));
    modal.show();
}

// Excluir conta
function excluirConta(id) {
    fetch(`/api/contas/${id}`, {
        method: 'DELETE'
    })
    .then(response => response.json())
    .then(data => {
        if (data.status === 'success') {
            // Fechar modal e recarregar contas
            const modal = bootstrap.Modal.getInstance(document.getElementById('modalConfirmacao'));
            modal.hide();
            
            carregarContas();
            carregarContasVencidas();
            carregarProximasContas();
            
            alert('Conta excluída com sucesso!');
        } else {
            alert('Erro: ' + data.message);
        }
    })
    .catch(error => {
        console.error('Erro ao excluir conta:', error);
        alert('Ocorreu um erro ao excluir a conta.');
    });
}

// Limpar formulário
function limparFormulario() {
    document.getElementById('conta-id').value = '';
    document.getElementById('conta-nome').value = '';
    document.getElementById('conta-valor').value = '';
    document.getElementById('conta-vencimento').valueAsDate = new Date();
    document.getElementById('conta-categoria').value = '';
    document.getElementById('conta-fixa').checked = false;
    document.getElementById('div-periodicidade').style.display = 'none';
    document.getElementById('conta-periodicidade').value = '';
    document.getElementById('conta-observacoes').value = '';
    
    // Resetar título do modal
    document.getElementById('modalContaLabel').textContent = 'Nova Conta';
}
